src/ctypes/ldouble.mli

```diff
@@ -0,0 +1,258 @@
+(*
+ * Copyright (c) 2016 Andy Ray.
+ *
+ * This file is distributed under the terms of the MIT License.
+ * See the file LICENSE for details.
+ *)
+
+type t
+(** The type of long doubles. *)
+
+val to_float : t -> float 
+(** Convert a long double to a float *)
+
+val of_float : float -> t 
+(** Create a long double from a float *)
+
+val to_int : t -> int 
+(** Convert a long double to an int *)
+
+val of_int : int -> t 
+(** Create a long double from an int *)
+
+val format : string -> t -> string
+(** This function is dangerous!  always use as "%<...>Lf" or related long double conversion! 
+    [format fmt ldbl] will be used in C as [sprintf(..., fmt, ldbl)]. *)
+
+val to_string : t -> string
+(** Convert a long double to a string *)
+
+val of_string : string -> t
+(** Create a long double from a string *)
+
+(* debugging only *)
+(*val to_hex_string : t -> string*)
+
+val add : t -> t -> t 
```

Comment:
If you're so inclined, it would be nice to have an `LDouble.Infix` module with aliases (`+`, `-`, etc.) for the arithmetic functions (and similarly for the `Complex` module).